### ISSUE TITLE
508-defect-2 [COGNITION]: WC Accordion - Expandable buttons should be wrapped in a heading with adjustable level

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#20485]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## User Story or Problem Statement

As an assistive technology user, I would like to be able to hear the accordion announced as headings.

## Details

Our content pages sometimes stack a number of accordions. Assistive tech needs a way to navigate them quickly, without having to parse through the entire page. Every screen reader user I've talked to or observed use headings as their primary navigational pattern.

Fundamentally these accordions are headings -- we are expanding content that is relational to the expandable trigger, so it helps organize our pages semantically to go with the visual weight of the UI.

## Acceptance Criteria

- [ ] Buttons are wrapped in a heading, defaulting to an H2
- [ ] Headings have a `level` prop that can be overridden to nest the heading properly
- [ ] Current styling and behavior are retained

## Proposed Solution (if known)
https://inclusive-components.design/collapsible-sections/ has a pretty good set of examples about wrapping components in headings.

```javascript
// va-accordion-item.tsx

// Defaults to an H2 in page
@Prop() level: number = 2;

render() {
  return (
    <Host>
      <h2 level={this.level}
        <button onClick={this.toggleOpen.bind(this)} aria-expanded={this.open ? 'true' : 'false'} aria-controls="content">
          {this.header}
        </button>
      </h2>
      <div id="content">
        <slot />
      </div>
    </Host>
  );
}
```

## WCAG or Vendor Guidance (optional)

* [Headings and Labels: Understanding SC 2.4.6](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-descriptive.html)
